crates/wasmparser/src/validator/operators.rs

```diff
@@ -310,1885 +380,3521 @@ impl OperatorValidator {
+    fn check_v128_relaxed_binary_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_relaxed_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] binary operator.
+    fn check_v128_unary_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] binary operator.
+    fn check_v128_funary_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_non_deterministic_enabled(offset)?;
+        self.check_v128_unary_op(offset)
+    }
+
+    /// Checks a [`V128`] binary operator.
+    fn check_v128_relaxed_unary_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_relaxed_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] relaxed ternary operator.
+    fn check_v128_relaxed_ternary_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_relaxed_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] relaxed ternary operator.
+    fn check_v128_bitmask_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::I32)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] relaxed ternary operator.
+    fn check_v128_shift_op(&mut self, offset: usize) -> OperatorValidatorResult<()> {
+        self.check_simd_enabled(offset)?;
+        self.stack.pop_operand(offset, Some(ValType::I32))?;
+        self.stack.pop_operand(offset, Some(ValType::V128))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    /// Checks a [`V128`] common load operator.
+    fn check_v128_load_op<T>(
+        &mut self,
+        offset: usize,
+        resources: &T,
+        memarg: MemoryImmediate,
+    ) -> OperatorValidatorResult<()>
+    where
+        T: WasmModuleResources,
+    {
+        self.check_simd_enabled(offset)?;
+        let idx = self.check_memarg(memarg, 3, offset, resources)?;
+        self.stack.pop_operand(offset, Some(idx))?;
+        self.stack.push_operand(offset, ValType::V128)?;
+        Ok(())
+    }
+
+    #[rustfmt::skip]
```

Comment:
How come this is here?

Reply:
The #[rustfmt::skip] greatly help to easily bulk transforme the code. I commonly used this throughout the lifetime of this PR and would only want to remove those annotations once we are absolutely sure to merge it.